frontend/src/scenes/surveys/surveyActivityDescriber.tsx

```diff
@@ -31,28 +42,102 @@ const surveyActionsMapping: Record<
+            return <strong>"{value.slice(0, 50)}..."</strong>
+        }
+
         return {
-            description: [<>updated the description</>],
+            description: [
+                <>
+                    updated the description from {formatDescription(change?.before as string | null | undefined)} to{' '}
+                    {formatDescription(change?.after as string | null | undefined)}
+                </>,
+            ],
         }
     },
     type: function onType(change) {
         return {
             description: [
                 <>
-                    changed the type to <strong>{change?.after as string}</strong>
+                    changed the type from <strong>{change?.before as string}</strong> to{' '}
+                    <strong>{change?.after as string}</strong>
                 </>,
             ],
         }
     },
-    questions: function onQuestions() {
+    questions: function onQuestions(change?: ActivityChange): ChangeMapping | null {
+        if (!change) {
+            return null
+        }
+
+        const beforeQuestions = change.before as Survey['questions']
+        const afterQuestions = change.after as Survey['questions']
+
+        if (beforeQuestions.length !== afterQuestions.length) {
+            return {
+                description: [
+                    <>
+                        changed the number of questions from <strong>{beforeQuestions.length}</strong> to{' '}
+                        <strong>{afterQuestions.length}</strong>
+                    </>,
+                ],
+            }
+        }
+
+        const questionChanges = afterQuestions
+            .map((afterQ, index) => {
+                const beforeQ = beforeQuestions[index]
+                if (JSON.stringify(beforeQ) !== JSON.stringify(afterQ)) {
+                    return {
+                        index: index + 1,
+                        changes: describeQuestionChanges(beforeQ, afterQ),
+                    }
+                }
+                return null
+            })
+            .filter((item): item is { index: number; changes: JSX.Element[] } => item !== null)
+
+        if (questionChanges.length === 0) {
+            return {
+                description: [<>No changes to questions</>],
+            }
+        }
+
         return {
-            description: [<>updated the questions</>],
+            description: [
+                <>
+                    updated <strong>{questionChanges.length}</strong> question{questionChanges.length !== 1 ? 's' : ''}:
+                    <ul className="bullet-list">
+                        {questionChanges.map(({ index, changes }) => (
+                            <li key={index}>
+                                Question {index}:
+                                <ul className="bullet-list">
+                                    {changes.map((changeItem, changeIndex) => (
+                                        <li key={changeIndex}>{changeItem}</li>
```

Comment:
this gives us a UI like this, which I'm pretty happy with (although the "Question 1" bit seems bit clunky, I just wanted to let them know which question was actually changed!)

<img width="772" alt="image" src="https://github.com/user-attachments/assets/bb8f3c84-3b5a-4bd9-921f-64a936dd9182">